### PR TITLE
Updating extended controls.

### PR DESCRIPTION
**Pull request by CptMitten**
Trying to match latest videodev2.h uapi.

### CODE REVIEW

**Comment by CptMitten** [comment]
Hello, not realy used to Github but I'm doing my best. 

I was trying to use extended controls with this binding, but I saw they were not fully implemented, so here is my update.

**Comment by CptMitten** [comment]
Hello @grega, just wanted to know if you saw my PR. If you have any question, do not hesitate.

File: v4l2.py [hunk 1]
```diff
@@ -1040,28 +1040,44 @@ class _u(ctypes.Union):
         _fields_ = [
             ('value', ctypes.c_int32),
             ('value64', ctypes.c_int64),
-            ('reserved', ctypes.c_void_p),
```
Comment (grega) [inline v4l2.py:1043]: Is there a reason to drop this particular field entirely?

Comment (CptMitten) [reply v4l2.py:1043]: I didn't really dropped it entirely, it was just renamed _ptr_ in order to match the kernel header naming.
Still a bit uncertain about necessity of the _ctype.POINTER()_ function, but it works fine with something like this:
```python
if qc.flags & v4l2.V4L2_CTRL_FLAG_HAS_PAYLOAD:
    ...
    ctrl.size = qc.elems * qc.elem_size
    val = (c_ubyte * ctrl.size)()
    ctrl.ptr = cast(val, POINTER(c_void_p))
    ...
ctrls.controls = pointer(ctrl)
ioctl(self.video_stream.fd, v4l2.VIDIOC_G_EXT_CTRLS, ctrls)
```

Comment (grega) [reply v4l2.py:1043]: Understood, thanks for clarifying 👍

**Comment by grega** [comment]
@davidplowman Would you mind casting your eyes over these changes?

**Comment by davidplowman** [comment]
Yes, this all looks good to me, and matching the official nomenclature make sense, so thanks very much for the update. I think it's good to keep updating this stuff, though in reality I'm sure it'll just be something we do as we discover the need. Anyway, I also ran the Picamera2 tests through on a Pi 4 (which uses this module) and there are no problems (I wouldn't really expect any), so thumbs up from me.

**Comment by grega** [comment]
Excellent, thanks @davidplowman and @CptMitten!

**Review by grega — APPROVED** [review]

**Comment by CptMitten** [comment]
Your welcome and thanks for your time. If you feel there is a need for a global update @davidplowman I might have a look. It shouldn't take me long.
Cheers.

**Pull request merged**